src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,73 @@
+package duke.task;
+
+import java.time.LocalDate;
+
+/**
+ * Class for Task object to encapsulate commonalities in subclasses.
+ *
+ * @author Andre Lin HuiKai
+ * @version CS2103T AY22/23 Semester 2
+ */
+public abstract class Task {
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Constructor for Task object.
+     * @param des Description of the task.
+     *
+     * @author Andre Lin HuiKai
+     * @version CS2103T AY22/23 Semester 2
+     */
+    public Task(String des) {
+        this.description = des;
+        this.isDone = false;
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    public void markTask() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmarks the task as incomplete.
+     */
+    public void unmarkTask() {
+        this.isDone = false;
+    }
+
+    /**
+     * Displays formatted information about the task.
+     * @return String containing common information about a task.
+     */
+    public String getStatusIcon() {
```

Comment:
I feel this method description is unnecessary as the class is already declared as an abstract class.

Reply:
Ah okay noted on this point